### PR TITLE
Organization management

### PR DESCRIPTION
**Pull request by atengler**
Allow filtering of requested resource lists by namespace and check if its possible to delete organization.

### CODE REVIEW

File: kqueen/blueprints/api/generic_views.py [hunk 1]
```diff
@@ -140,15 +148,34 @@ class ListView(GenericView):
     action = 'list'
 
     def set_object(self, *args, **kwargs):
+        if request.args.get('all_namespaces'):
```
Comment (tomkukral) [inline kqueen/blueprints/api/generic_views.py]: Why `args`? I'd expect it to be `kwarg` - `namespace='all'`

File: kqueen/blueprints/api/generic_views.py [hunk 2]
```diff
@@ -140,15 +148,34 @@ class ListView(GenericView):
...
 
     def get_content(self, *args, **kwargs):
+        if request.args.get('all_namespaces'):
+            objs = []
+            for obj in self.obj:
+                namespace = obj._object_namespace
+                obj_dict = obj.get_dict(expand=True)
+                obj_dict['namespace'] = namespace
+                objs.append(obj_dict)
+            return objs
```
Comment (tomkukral) [inline kqueen/blueprints/api/generic_views.py]: We are returning:

* object for single namespace
* list of dicts (not objects) for all_namespaces

It this desired behavior? Why do we don't return list of object?*

* and add _namespace property to objects?

File: kqueen/models.py [hunk 3]
```diff
@@ -376,6 +376,44 @@ class Organization(Model, metaclass=ModelMeta):
     policy = JSONField()
     created_at = DatetimeField(default=datetime.utcnow)
 
+    def is_deletable(self):
+        remaining = []
+        if User.list(self.namespace, return_objects=False):
+            all_users = User.list(None).values()
+            users = [u for u in all_users if u.namespace == self.namespace]
```
Comment (tomkukral) [inline kqueen/models.py]: not a blocker: I'd prefer to use more lines of code and make this code more readable.

File: kqueen/models.py [hunk 4]
```diff
@@ -376,6 +376,44 @@ class Organization(Model, metaclass=ModelMeta):
...
+                })
+        if remaining:
+            return False, remaining
+        return True, remaining
+
+    def delete(self):
+        deletable, remaining = self.is_deletable()
+        if deletable:
+            return super().delete()
+        resources = ','.join(['{} {}'.format(r['object'], r['uuid']) for r in remaining])
```
Comment (tomkukral) [inline kqueen/models.py]: This is hard to read, can you make this less magical?

File: kqueen/blueprints/api/generic_views.py (hunk 1 above)
Comment (atengler) [reply kqueen/blueprints/api/generic_views.py]: List is a GET only endpoint, so the `all_namespaces` param is passed via urlencode and appears as arg on request

File: kqueen/blueprints/api/generic_views.py (hunk 2 above)
Comment (atengler) [reply kqueen/blueprints/api/generic_views.py]: I agree, that we could prefix the namespace attribute, so it does not look as default part of the schema for given resource, but I don't think whether its object or dict at this point, I think it gets serialized in the exactly same way either way before its returned.

File: kqueen/models.py (hunk 3 above)
Comment (atengler) [reply kqueen/models.py]: In this case I can't really see how to make this even more readable :D.

File: kqueen/models.py (hunk 4 above)
Comment (atengler) [reply kqueen/models.py]: This one could get more readable, true that. I'll make it so

File: kqueen/blueprints/api/generic_views.py (hunk 2 above)
Comment (tomkukral) [reply kqueen/blueprints/api/generic_views.py]: Thanks, waiting for fix.

File: kqueen/models.py (hunk 3 above)
Comment (tomkukral) [reply kqueen/models.py]: This on is acceptable.

**Pull request merged**